qiskit/algorithms/minimum_eigensolvers/minimum_eigensolver.py

```diff
@@ -0,0 +1,98 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The minimum eigensolver interface and result."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ..algorithm_result import AlgorithmResult
+from ..list_or_dict import ListOrDict
+
+
+class MinimumEigensolver(ABC):
+    """The minimum eigensolver interface.
+
+    Algorithms that can compute a minimum eigenvalue for an operator may implement this interface to
+    allow different algorithms to be used interchangeably.
+    """
+
+    @abstractmethod
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> "MinimumEigensolverResult":
+        """
+        Computes the minimum eigenvalue. The ``operator`` and ``aux_operators`` can be supplied here
+        and if not ``None`` will override any already set into algorithm so it can be reused with
+        different operators. While an ``operator`` is required by algorithms, ``aux_operators`` are
+        optional.
+
+        Args:
+            operator: Qubit operator of the observable.
+            aux_operators: Optional list of auxiliary operators to be evaluated with the
+                parameters of the minimum eigenvalue main result and their expectation values
+                returned. For instance in chemistry these can be dipole operators and total particle
+                count operators, so we can get values for these at the ground state.
+
+        Returns:
+            A minimum eigensolver result.
+        """
+        return MinimumEigensolverResult()
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        """Whether computing the expectation value of auxiliary operators is supported.
+
+        If the minimum eigensolver computes an eigenvalue of the main ``operator`` then it can
+        compute the expectation value of the ``aux_operators`` for that state. Otherwise they will
+        be ignored.
+
+        Returns:
+            True if aux_operator expectations can be evaluated, False otherwise
+        """
+        return False
+
+
+class MinimumEigensolverResult(AlgorithmResult):
+    """Minimum eigensolver result."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._eigenvalue = None
+        self._aux_operator_eigenvalues = None
+
+    @property
+    def eigenvalue(self) -> complex | None:
+        """The computed minimum eigenvalue."""
+        return self._eigenvalue
+
+    @eigenvalue.setter
+    def eigenvalue(self, value: complex) -> None:
+        self._eigenvalue = value
```

Comment:
I removed the docstring on the setter just because it's ignored by Sphinx, so the user will only ever see the docstring on the property. I can add it back if you think people looking through the code might benefit from it?

Reply:
Fair enough, I can just remove it on my side then.

Reply:
To be fair, I haven't been consistent with this. I still have the setter docstring in a VQE for example... I'll change these.